data/eigerfan/src/EigerFan.cpp

```diff
@@ -385,7 +389,18 @@ void EigerFan::HandleStreamMessage(zmq::message_t &message, boost::shared_ptr<zm
           lastFrameSent = frame;
         }
         num_frames_sent++;
+        if (currentConsumerIndexToSendTo < num_frames_consumed.size()) {
+          num_frames_consumed[currentConsumerIndexToSendTo]++;
+        }
+        else {
+          LOG4CXX_WARN(log, "Error counting consumer frames for logging");
+        }
       } else if (htype.compare(END_HEADER_TYPE) == 0) {
+        LOG4CXX_INFO(log, "End of series message received after " + boost::lexical_cast<std::string>(num_frames_sent) \
+                + " frames sent");
+        for(int j=0; j<num_frames_consumed.size(); j++)
+          LOG4CXX_INFO(log, "Consumer index " + boost::lexical_cast<std::string>(j) \
```

Comment:
Wondering about the log message inside a for-loop... How about generating a (long) string in the for-loop and then just logging once with the content of that string?